src/components/CompareResults/CommonGraph.tsx

```diff
@@ -0,0 +1,210 @@
+import { Chart as ChartJS, LineElement, LinearScale, TooltipItem } from 'chart.js';
+import 'chart.js/auto';
+import { Line } from 'react-chartjs-2';
+import { style } from 'typestyle';
+
+import { Spacing } from '../../styles';
+
+ChartJS.register(LinearScale, LineElement);
+
+interface GraphRun {
+  from: string;
+  oxValue: number;
+  value: number;
+}
+interface GraphData {
+  x: number;
+  y: number | string;
+}
+
+const configGraph = () => {
+  const options = {
+    plugins: {
+      datalabels: {
+        formatter: function (value: GraphData) { return value.y || null;  },
+      },
+      legend: {
+        align: 'start' as const,
+        position: 'top' as const,
+      },
+      tooltip: {
+        callbacks: {
+          label: (context: TooltipItem<'line'>) => {
+            return `${(context.raw as GraphData).y}`;
+          },
+        },
+      },
+      title: {
+        display: true,
+      },
+    },
+    scales: {
+      x: {
+        grid: {
+          display: false,
+          offset: false,
+        },
+      },
+      y: {
+        ticks: {
+          display: false,
+          beginAtZero: true,
+        },
+        grid: {
+          drawBorder: false,
+          display: false,
+          offset: false,
+        },
+      },
+    },
+  };
+  return { options };
+};
+
+const initializeGraph = (graphWidth: number) => {
+  const graph = new Array(graphWidth) as GraphRun[];
+
+  for (let index = 0; index < graphWidth; index++) {
+    const initialObject = {
+      from: '',
+      oxValue: 0,
+      value: 0,
+    };
+    graph[index] = initialObject;
+  }
+
+  return graph;
+};
+
+const addToHist = (graph: GraphRun[], X: GraphRun, index: number) => {
+  graph[index].from = X.from;
+  graph[index].value += Math.exp(-(((index - X.value) / 10) ** 2));
+};
+
+const formatDataset = (
+  values: GraphRun[],
+  maxX: number,
+  graph: GraphRun[],
+  graphWidth: number,
+) => {
+  values.forEach((x, index) => {
+    addToHist(
+      graph,
+      { from: x.from, oxValue: x.value, value: (x.value * graphWidth) / maxX },
```

Comment:
I don't understand the computation for `value`, it would be good to have some explanation about it.

Reply:
Also removed from the current version.